force-app/main/default/classes/tasks/CollectionsTasks.cls

```diff
@@ -52,6 +73,16 @@ public class CollectionsTasks {
     List<String> codeTicks,
     List<Integer> priceTicks
   ) {
-    return 0;
+
+    Integer i = 0;
+    Integer stockPosition = 0;
+    for (String stockName : codeTicks) {
```

Comment:
Imagine the input size is 100 000.
The stock price we are interested in is met only couple of times in the array.

How we can make the algorithm more efficient?

Reply:
Well. I've looked at several possible solutions:
1. Reverse the list and take the first value with indexOf(). Doesn't work because we don't have reverse method in apex;
2. take the first value with indexOf(), then get rid of all previous values and try to get value again with indexOf. Doesn't work because we don't have truncate method in apex;
3. take the first value with indexOf(), then use indexOf() again starting with the index of previous value. Doesn't work because indexOf() takes only one parameter.

So for me the only possible optimisation is that I've done.

Reply:
try checking the collection from the end and not the beginning?

Reply:
Updated, thanks.